horace_core/sqw/PixelData/@PixelData/append.m

```diff
@@ -0,0 +1,62 @@
+function obj = append(obj, pix)
+% Join the pixels in the given PixelData object to the end of this
+% object.
+%
+% The pixels to append must all be in memory and you cannot append pixels if
+% the inputted PixelData object has more pixels than are allowed in a single
+% page.
+%
+% Input
+% -----
+% pix    A PixelData object containing the pixels to append
+%
+if ~isa(pix, 'PixelData')
+    error('PIXELDATA:append', ...
+            'Input object must be of class ''PixelData''. Found ''%s''', ...
+            class(pix));
+end
+if isempty(pix)
+    return;
+end
+if pix.num_pixels > obj.max_page_size_
+    error('PIXELDATA:append', ...
+            ['Cannot append more pixels than allowed in a single page\n ' ...
+            'Found ''%i'' pixels, ''%i'' allowed.'], pix.num_pixels, ...
+            obj.max_page_size_);
+end
+
+if ~(obj.get_num_pages_() == obj.page_number_)
+    if obj.page_is_dirty_(obj.page_number_) && obj.dirty_page_edited_
+        obj.write_dirty_page_();
+    end
+    obj.load_page_(obj.get_num_pages_());
+end
+
+if isempty(obj)
+    obj.data_ = pix.data;
+    obj.set_page_dirty_(true);
+else
```

Comment:
does the non-empty case need to be a nested `if` rather than `if ... elsif ... else` ?

that would make the three cases we are handling more explicit:
- current object is empty
- current page is full (=> save and create a new page)
- current page is partially full (=> fill current page, save, put remaining pixels in new page)